### DISCUSSION TITLE
How respective rules are loaded when stream gets payload

### DISCUSSION THREAD

**Discussion by sujith1994** [lf-edge/ekuiper#1257] [General]
For instance, I have a stream called demo and there are 2 rules created for the same (rule1,rule2).
When i send data to "demo",  how rule engine understand this time  rule1,rule2 should execute AND how rule is loaded from db ?

I had go through the source code to understand , since i'm not that good at golang , i didn't get the understanding well.

**Comment by ngjaying** [comment]
The stream definition is basically a "logical" representation of the data source. When a rule started, it will connect to the data source according to this "logical definition". If want multiple rules to process the exact same stream of data, you can define the stream to be shared which will create one instance of data source and distribute to these rules.

Please check https://ekuiper.org/docs/en/latest/concepts/sources/overview.html
